image_view/src/video_recorder.cpp

```diff
@@ -0,0 +1,40 @@
+/****************************************************************************
+* Software License Agreement (Apache License)
+*
+*     Copyright (C) 2012-2013 Open Source Robotics Foundation
+*
+*     Licensed under the Apache License, Version 2.0 (the "License");
+*     you may not use this file except in compliance with the License.
+*     You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+*     Unless required by applicable law or agreed to in writing, software
+*     distributed under the License is distributed on an "AS IS" BASIS,
+*     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*     See the License for the specific language governing permissions and
+*     limitations under the License.
+*
+*****************************************************************************/
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <memory>
+
+#include "image_view/video_recorder_node.hpp"
+
+int main(int argc, char** argv)
+{
+  using image_view::VideoRecorderNode;
+
+  rclcpp::init(argc, argv);
+
+  rclcpp::NodeOptions options;
+  auto vr_node = std::make_shared<VideoRecorderNode>(options);
+
+  rclcpp::spin(vr_node);
+
+  std::cout << "\nVideo saved as " << vr_node->filename << std::endl;
```

Comment:
Is this necessary? shouldn't it be in the node destructor vs in the `main()`

Reply:
Moved to destructor.